meson_cpu/x86/meson.build

```diff
@@ -1,228 +1,102 @@
-)
-FMA4 = mod_features.new(
-  'FMA4', 22, implies: AVX, args: '-mfma4',
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_fma4.c')[0]
-)
-# x86 half-precision
-F16C = mod_features.new(
-  'F16C', 23, implies: AVX, args: '-mf16c',
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_f16c.c')[0]
-)
-FMA3 = mod_features.new(
-  'FMA3', 24, implies: F16C, args: '-mfma',
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_fma3.c')[0]
-)
-# match this to HWY_AVX2
-AVX2 = mod_features.new(
-  'AVX2', 25, implies: FMA3, args: ['-mavx2', '-maes', '-mpclmul', '-mbmi', '-mbmi2'],
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx2.c')[0]
-)
-# 25-40 left as margin for any extra features
-AVX512F = mod_features.new(
-  'AVX512F', 40, implies: [AVX2],
-  # Disables mmx because of stack corruption that may happen during mask
-  # conversions.
-  # TODO (seiko2plus): provide more clarification
-  args: ['-mno-mmx', '-mavx512f'],
-  detect: {'val': 'AVX512F', 'match': '.*'},
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512f.c')[0],
-  extra_tests: {
-    'AVX512F_REDUCE': files(source_root + '/numpy/distutils/checks/extra_avx512f_reduce.c')[0]
-  }
-)
-AVX512CD = mod_features.new(
-  'AVX512CD', 41, implies: AVX512F, args: '-mavx512cd',
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512cd.c')[0]
-)
-AVX512_KNL = mod_features.new(
-  'AVX512_KNL', 42, implies: AVX512CD, args: ['-mavx512er', '-mavx512pf'],
-  group: ['AVX512ER', 'AVX512PF'],
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512_knl.c')[0]
-)
-AVX512_KNM = mod_features.new(
-  'AVX512_KNM', 43, implies: AVX512_KNL,
-  args: ['-mavx5124fmaps', '-mavx5124vnniw', '-mavx512vpopcntdq'],
-  group: ['AVX5124FMAPS', 'AVX5124VNNIW', 'AVX512VPOPCNTDQ'],
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512_knm.c')[0]
-)
-AVX512_SKX = mod_features.new(
-  'AVX512_SKX', 50, implies: AVX512CD,
-  args: ['-mavx512vl', '-mavx512bw', '-mavx512dq'],
-  group: ['AVX512VL', 'AVX512BW', 'AVX512DQ'],
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512_skx.c')[0],
-  extra_tests: {
-    'AVX512BW_MASK': files(source_root + '/numpy/distutils/checks/extra_avx512bw_mask.c')[0],
-    'AVX512DQ_MASK': files(source_root + '/numpy/distutils/checks/extra_avx512dq_mask.c')[0]
-  }
-)
-AVX512_CLX = mod_features.new(
-  'AVX512_CLX', 51, implies: AVX512_SKX, args: '-mavx512vnni',
-  group: ['AVX512VNNI'],
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512_clx.c')[0]
-)
-AVX512_CNL = mod_features.new(
-  'AVX512_CNL', 52, implies: AVX512_SKX,
-  args: ['-mavx512ifma', '-mavx512vbmi'],
-  group: ['AVX512IFMA', 'AVX512VBMI'],
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512_cnl.c')[0]
-)
+HWY_SSE4_FLAGS = ['-DHWY_WANT_SSE4', '-DHWY_DISABLE_PCLMUL_AES']
+# Use SSE for floating-point on x86-32 to ensure numeric consistency.
```

Comment:
This fix needs to be backported too.

Reply:
It will be backported in #30026, and yes, I wonder if you have encountered such ambiguous runtime floating-point errors on 32-bit builds?